### PR TITLE
Add basic parallel processing

### PR DESCRIPTION
**Pull request by AnthonyLim23**
This PR does 2 things:

1. Adds the basic tools to do parallel processing
2. Pins Cython see https://github.com/mantidproject/quickBayes/issues/70

The new parallel function just takes a counter. This is good for our purpose as the idea is to use it to parallelise over the workspace index. 

To check the docs click details on the read the docs build.

### CODE REVIEW

File: docs/source/dev/parallel.rst [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+           partial_function = partial(self.function, constant=constant)
+           return parallel(values, partial_function)
+
+   inputs = [ (k, -k) for k in range(10)]
+   const = 3.
+   test = test_parallel(2.)
+   result = test.parallel_function(inputs, const)
+   print(result)
+
+It is important to note that the method call is to a wrapper of themethod function we want to run in parallel.
```
Comment (RichardWaiteSTFC) [inline docs/source/dev/parallel.rst:89]: Typo `the method`

File: setup.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 from tools.setup_helper import get_extensions
 
 
-VERSION = "1.0.0b10"
+VERSION = "1.0.0b12"
```
Comment (RichardWaiteSTFC) [inline setup.py:5]: Incremented version by 2, or am I missing something?

File: src/quickBayes/utils/parallel.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    Use threads as the default does not work with Mantid.
+    :input items: the list to loop over
+    :input function: the function to run in parallel
+    :input N: the number of process to use
+    :return a list of the outputs from function. If multuple outputs
+    from function then the first index is for the loop value and the
+    second index is for the item from function.
+    """
+    return Parallel(n_jobs=N,
+                    prefer="threads")(delayed(function)(j) for j in items)
```
Comment (RichardWaiteSTFC) [inline src/quickBayes/utils/parallel.py:21]: Is there any reason not to specify `backend=threading`?

**Review by RichardWaiteSTFC — CHANGES_REQUESTED** [review]
Looks good, thanks for the nice documentation (one small typo sorry!). It seems to be a very versatile way of using multiple threads to parallelise loops.

I had a question about the use of prefer over hard-coding the backend - but that's just for my understanding!

File: docs/source/dev/parallel.rst (hunk 1 above)
Comment (AnthonyLim23) [reply docs/source/dev/parallel.rst:89]:
```suggestion
It is important to note that the method call is to a wrapper of the method function we want to run in parallel.
```

File: setup.py (hunk 2 above)
Comment (AnthonyLim23) [reply setup.py:5]: I pushed 2 versions (the first was using the default and the second was using threading)

File: src/quickBayes/utils/parallel.py (hunk 3 above)
Comment (AnthonyLim23) [reply src/quickBayes/utils/parallel.py:21]: I dont really understand it (not looked into it in detail). But the default is to not use threading and instead it does something with the Python console (thats why it doesnt work in mantid)

**Pull request merged**